internal/storage/storage.go

```diff
@@ -210,48 +202,56 @@ func (db *Database) GetRows(query string) (pgx.Rows, error) {
 	return rows, nil
 }
 
-func (db *Database) Exec(query string) (pgconn.CommandTag, error) {
-	res, err := db.ConnPool.Exec(db.CTX, query)
+func (db *Database) Exec(ctx context.Context, query string) (pgconn.CommandTag, error) {
+	res, err := db.ConnPool.Exec(ctx, query)
 	if err != nil {
 		return nil, err
 	}
 	return res, nil
 }
 
-func (db *Database) GetDBConnection() (*pgxpool.Pool, error) {
-	pool, err := pgxpool.Connect(db.CTX, db.DBConnURL)
+func (db *Database) GetDBConnection(ctx context.Context) (*pgxpool.Pool, error) {
+	pool, err := pgxpool.Connect(ctx, db.DBConnURL)
 	if err != nil {
 		return nil, err
 	} else {
 		return pool, nil
 	}
 }
 
-func (db *Database) Ping() error {
+func (db *Database) Ping(ctx context.Context) error {
 	if db.DBErrorConnect != nil {
 		return db.DBErrorConnect
 	} else {
-		err := db.ConnPool.Ping(db.CTX)
+		err := db.ConnPool.Ping(ctx)
 		return err
 	}
 }
 
-func (db *Database) AddURL(url string, user string) (string, error) {
+func (db *Database) AddURL(ctx context.Context, url string, user string) (string, error) {
 	var newID int64
 
-	row := db.ConnPool.QueryRow(db.CTX, "INSERT INTO public.storage (full_url, user_id) VALUES ($1, $2) RETURNING id",
-		url, user)
+	row := db.ConnPool.QueryRow(ctx,
+		"INSERT INTO public.storage (full_url, user_id) VALUES ($1, $2) RETURNING id", url, user)
 	if err := row.Scan(&newID); err != nil {
-		return "", fmt.Errorf("318: row.Scan: %w", err)
+		id, err := db.SearchID(ctx, url)
+		if err == nil {
+			return db.BaseURL + strconv.Itoa(id), middleware.ErrConflict
+		} else {
+			return "", middleware.ErrConflict
+		}
 	} else {
 		return db.BaseURL + strconv.FormatInt(newID, 10), nil
 	}
 }
 
-func (db *Database) SearchURL(id int) (string, error) {
+func (db *Database) SearchURL(ctx context.Context, id int) (string, error) {
 	var url string
-	query := fmt.Sprintf("select full_url from %s.%s where id = %v", schema, table, id)
-	row, err := db.GetRows(query)
+	//query := fmt.Sprintf("select full_url from $1.$2 where id = $3", schema, table, id)
+	//log.Println(query)
+	//row, err := db.GetRows(ctx, query)
+	row, err := db.ConnPool.Query(ctx, "select full_url from public.storage where id = $1", id)
```

Comment:
Поскольку здесь дергаем одну строку, то можно сделать так:
```suggestion
	err := db.ConnPool.QueryRow(ctx, "select full_url from public.storage where id = $1", id).Scan(&url)
```
Если же оставить Query, то после цикла `for row.Next()` необходима еще проверка `if err := row.Err(); err != nil `

Tutorial: https://go.dev/doc/database/querying